picard/ui/itemviews.py

```diff
@@ -250,40 +269,44 @@ def create_icons(self):
         ]
         self.icon_plugins = icontheme.lookup('applications-system', icontheme.ICON_SIZE_MENU)
 
-    def update_selection(self, i, j):
-        self._selected_view = i
-        self.views[j].clearSelection()
-        self.window.update_selection(
-            [item.obj for item in self.views[i].selectedItems()])
-
-    def update_selection_0(self):
-        if not self._ignore_selection_changes:
-            self._ignore_selection_changes = True
-            self.update_selection(0, 1)
-            self._ignore_selection_changes = False
-
-    def update_selection_1(self):
-        if not self._ignore_selection_changes:
-            self._ignore_selection_changes = True
-            self.update_selection(1, 0)
-            self._ignore_selection_changes = False
+    def _update_selection(self, selected_view):
+        for view in self._views:
+            if view != selected_view:
+                view.clearSelection()
+            else:
+                self._selected_view = view
+                self.window.update_selection([item.obj for item in view.selectedItems()])
 
     def update_current_view(self):
-        self.update_selection(self._selected_view, abs(self._selected_view - 1))
+        self._update_selection(self._selected_view)
 
     def remove(self, objects):
         self._ignore_selection_changes = True
         self.tagger.remove(objects)
         self._ignore_selection_changes = False
 
-        view = self.views[self._selected_view]
+        view = self._selected_view
         index = view.currentIndex()
         if index.isValid():
             # select the current index
             view.setCurrentIndex(index)
         else:
             self.update_current_view()
 
+    def set_sorting(self, sort=True):
+        # Add 1 if sorting is disabled, subtract 1 if sorting is enabled
+        with self._sorting_lock:
+            self._sorting_count += -1 if sort else 1
+
+        # Guarantee counter is >= 0
+        self._sorting_count = 0 if self._sorting_count < 0 else self._sorting_count
+
+        # If counter == 0, re-enable sorting
+        sort = True if self._sorting_count == 0 else False
+        if self._views[0].isSortingEnabled() != sort:
```

Comment:
This test looks weird to me. Why is it needed?

Reply:
It isn't. Just skipping the other calls if the panel sorting matches the sort flag. No need to disable something that already is disabled, or enable something that already is enabled. Didn't really test if it triggers sorting on the second case.

Reply:
I'd rather simplify, and get rid of the test.

```python
            for view in self._views:
                view.setSortingEnabled(sort)
```

Reply:
It does trigger a re-sorting. Measured the times:

1. Load
- disable empty panel: 0ms
- enable panel with 19k files: 185ms

1. Clustering
- disable panel with 19k files: 0ms
- enable panel with 19k files + 1.9k clusters: 70ms